packages/amplify-graphql-transformer-core/src/utils/strip-directives.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  ObjectTypeDefinitionNode,
+  DirectiveNode,
+  InterfaceTypeDefinitionNode,
+  UnionTypeDefinitionNode,
+  ScalarTypeDefinitionNode,
+  InputObjectTypeDefinitionNode,
+  FieldDefinitionNode,
+  InputValueDefinitionNode,
+  EnumValueDefinitionNode,
+  EnumTypeDefinitionNode,
+  Kind,
+  DocumentNode,
+} from 'graphql';
+
+export function stripDirectives(doc: DocumentNode, except: string[] = []): DocumentNode {
+  const definitions = [];
+  for (const def of doc.definitions) {
+    switch (def.kind) {
+      case Kind.OBJECT_TYPE_DEFINITION:
```

Comment:
There's lots of places in this PR where you use this pattern for calling a different method for each enum type or something similar. IMO, it's cleaner to define a map like stripDefMap: Record<Kind, Function> = {...} and then use it like
```js
doc.definitions.forEach(def => {
  definitions.push(stripDefMap[def.kind](def))
}
```
And if you want to have default behavior, you can use lodash _.get() when indexing the map:
```js
_.get(stripDefMap, [def.kind], () => { default behavior here })
```

There are some cases where this approach doesn't make sense like when you want to apply the same function in multiple cases, but I think a lot of these switch statement patterns could be made much DRYer with this approach